ops/model.py

```diff
@@ -1767,6 +1774,136 @@ def remote_model(self) -> RemoteModel:
             self._remote_model = RemoteModel(uuid=d['uuid'])
         return self._remote_model
 
+    def load(
+        self,
+        cls: type[_T],
+        app_or_unit: Unit | Application,
+        *args: Any,
+        decoder: Callable[[str], Any] | None = None,
+        errors: Literal['raise', 'blocked'] = 'raise',
+        **kwargs: Any,
+    ) -> _T:
+        """Load the data for this relation into an instance of a data class.
+
+        The raw Juju relation data is passed to the data class's ``__init__``
+        method as keyword arguments, with values decoded using the provided
+        decoder function, or :func:`json.loads` if no decoder is provided.
+
+        Any additional positional or keyword arguments will be passed through to
+        the data class ``__init__``.
```

Comment:
Might be good to have a tiny example of usage here? Don't even need to have the data class definition. Something like:

```
Example::

    relation = self.model.get_relation('db')
    data = relation.load(DatabaseModel, self.app)
    secret_id = data.credentials
```